app/src/main/java/com/google/android/soundchecker/BaseFilePlayerActivity.kt

```diff
@@ -339,4 +350,30 @@ open class BaseFilePlayerActivity : ComponentActivity(), OnAudioFocusChangeListe
             }
         }
     }
+
+    inner class PlaybackStateChangedReceiver : BroadcastReceiver() {
+        override fun onReceive(context: Context, intent: Intent) {
+            when (intent.action) {
+                FilePlayerService.ACTION_START_PLAYBACK -> {
+                    sendStartPlaybackMsg(0)
+                }
+
+                FilePlayerService.ACTION_STOP_PLAYBACK -> {
+                    sendStopPlaybackMsg(0)
+                }
+
+                else -> {
+                    // Unknown action, ignore
+                }
+            }
+        }
+    }
+
+    private fun checkAndRequestNotificationPermission() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU &&
+            checkSelfPermission(POST_NOTIFICATIONS) != PackageManager.PERMISSION_GRANTED) {
+                registerForActivityResult(
+                        ActivityResultContracts.RequestPermission()) { }.launch(POST_NOTIFICATIONS)
```

Comment:
Do you get a callback for this? If so, can you make confirm that this code works both when the permission is already granted and when you need to wait for the notification?

Reply:
Yes.